### DISCUSSION TITLE
Good practice to release the game.

### DISCUSSION THREAD

**Discussion by ghost** [mgrojo/ASFML#23] [General]
Hi!

First of all, I would like to thank you for your work on Ada. I discovered Alire yesterday, and then your binding : I am amazed by your dedication. 
I coded a lot with Ada in the past years and as I want to make a game on my spare time, why not use ASFML instead of unity ;). 

I am not used to produce installation wizard or stuff like that, and I am wondering do you have any advice to do some packaging of the executable and dll etc. Alire actually hides a lot of thing with its dependencies management. (I currently work on windows to avoid to configure cross compilation, as I target windows, which may be to complicated for me.) 

Thx.

R-L.

**Comment by mgrojo** [comment]
Hi @Rose-Lilas,

Glad you find ASFML useful.

Regarding packaging, one way to distribute an Ada application is to [include it in the Alire index](https://alire.ada.dev/docs/publishing.html), because users will be able to get all needed it to build it automatically using `alr get --build your_game`. Another option, unfortunately only for Linux, is to build an AppImage of the game. You can see an example of both options in my simple game [play_2048](https://github.com/mgrojo/play_2048).

For Windows, an option that might work is to use the new feature in Alire 2.0 using the [`alr install`](https://github.com/alire-project/alire/blob/master/doc/user-changes.md#installation-of-local-crates) command and then packaging the resulting directory with all the dependencies. I haven't tested it, but I guess it should work, except that maybe you have to add there also the external dependencies like CSFML/SFML or instruct users on how to install it. Maybe @HonkiTonk can advise on that, since he makes Windows releases for his game [HonkiTonks Civilizationen](https://github.com/HonkiTonk/Honki-Tonks-Zivilisationen/releases).

**Reply by HonkiTonk** [reply]
Even if my reply is a little late (I've had a few health problems over the last few months), I'm writing it anyway.

The Alire version contains the source code and all files that my game loads without the executable and without CSFML/SFML files. On Windows you have to add the corresponding DLL files (maybe I should add them to GitHub/Alire too?) and on Linux you have to install the CSFML/SFML/GLIBC via the package manager to be able to play my game.

The versions I publish on GitHub are similar. Only without the source code, but with the executable file and, in the case of the Windows package, also the DDL files.

I don't use a crosscompiler either, even if one is available under Linux. For Linux I use GNAT, installed via the package manager, and for Windows I use the Windows version of GNAT via WINE, so I can test more easily whether everything works system-related or not. I don't know if that helps or if I'm just talking past the problem. Maybe I should create a little tutorial on how I always put it together.

**Comment by ghost** [comment]
Thank you for the answer. I will check your proposals. 

Just for my information, where are stored resources like 'cute_image.jpg' that you refer in your example (main.adb)? 
Maybe you can add in the readme that the example is not standalone and that user must provide the resources.

**Reply by mgrojo** [reply]
Yes, the user must provide the resources. You're right about that this should be stated somewhere. I take note to fix that at some point.

**Reply by mgrojo** [reply]
I've added appropriate comments and made it resemble better the original example of SFML.
